File: Java Blind 75/String/Valid Anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        // Time: O(n) Space: O(1)
        if (s.length() != t.length()) return false; // anagrams are same lengths

        int[] count = new int[26]; // 26 letters in alphabet
        for (int i = 0; i < s.length(); i++) {
            count[s.charAt(i) - 'a']++; // increments the count of the letter at index
            count[t.charAt(i) - 'a']--; // decrements the count of the letter at index
        }
        for (int i = 0; i < count.length; i++) {
            if (count[i] != 0) return false; // checks if they had equal amounts of letters
        }
        return true;
    }
}

File: Python Blind 75/String/Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # Time: O(n), Space O(1)
        if len(s) != len(t): return False # If the lengths are not the same, they can't be anagrams

        count = [0] * 26 # creates an array of twenty six 0's which will count the number of chars
        for i in range(len(s)):
            count[ord(s[i]) - ord('a')] += 1
            count[ord(t[i]) - ord('a')] -= 1
        return count == [0] * 26